tests/falcon_test.py

```diff
@@ -88,3 +119,32 @@ class HelloResource:
         spec.path(resource=hello_resource)
 
         assert spec._paths["/hi"]["x-extension"] == "global metadata"
+
+    def test_unredundant_basepath_resource(self, app, spec_factory, settings):
+        class HelloResource:
+            def on_get(self, req, resp):
+                """A greeting endpoint.
+                ---
+                description: get a greeting
+                responses:
+                    200:
+                        description: said hi
+                """
+                return "dummy"
+
+        expected = {
+            "description": "get a greeting",
+            "responses": {"200": {"description": "said hi"}},
+        }
+        full_path = "/v1/foo/v1"
+        path_expected = "/foo/v1"
+        hello_resource = HelloResource()
+        app.add_route(full_path, hello_resource)
+        spec = spec_factory(app)
+        base_path = \
+            settings["servers"][0]["variables"]["basePath"].get("default")
```

Comment:
Is this how you would normally specify the base path? When possible, I'd like tests to mirror actual usage.

The `settings` fixture is adding unnecessary complexity. Do you mind reverting?

Reply:
Well, since openapi 3.x.x configuration allow multiple servers the basePath resides deep within each server tree, unlike swagger 2 that puts it in root tree.